example/examples/CacheURLTiles.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { StyleSheet, View, Text, Dimensions } from 'react-native';
+
+import MapView, {
+  MAP_TYPES,
+  PROVIDER_DEFAULT,
+  ProviderPropType,
+  UrlTile,
+} from 'react-native-maps';
+
+const { width, height } = Dimensions.get('window');
+
+const ASPECT_RATIO = width / height;
+const LATITUDE = 37.78825;
+const LONGITUDE = -122.4324;
+const LATITUDE_DELTA = 0.0922;
+const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;
+
+class CustomTiles extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      region: {
+        latitude: LATITUDE,
+        longitude: LONGITUDE,
+        latitudeDelta: LATITUDE_DELTA,
+        longitudeDelta: LONGITUDE_DELTA,
+      },
+    };
+  }
+
+  get mapType() {
+    // MapKit does not support 'none' as a base map
+    return this.props.provider === PROVIDER_DEFAULT
+      ? MAP_TYPES.STANDARD
+      : MAP_TYPES.NONE;
+  }
+
+  render() {
+    const { region } = this.state;
+    return (
+      <View style={styles.container}>
+        <MapView
+          provider={this.props.provider}
+          mapType={this.mapType}
+          style={styles.map}
+          initialRegion={region}
+        >
+          <UrlTile
+            urlTemplate="https://stamen-tiles.a.ssl.fastly.net/toner/{z}/{x}/{y}.png"
+            zIndex={-1}
+            // Test steps:
+            // 1) Without new tile provider properties: comment out tileCachePath & maximumNativeZ
+            // 2) With maximumNativeZ only to test scaling past maxNativeZoom level
+            // 3) With doubleTileSize (only on Android)
+            // 4) With tileCachePath too - test caching performance with cutting & throttling network connectivity
+            // 5) With tileCacheMaxAge too
+            // 6) With offlineMode=true too - test zoom in to test scaling of lower zoom level tiles to higher zoom levels
+            //
+            maximumNativeZ={15}
+            // For testing activate different tile cache paths, examples below
+            // work for simulator / emulator testing
+            // This is for iOS simulator
+						//tileCachePath="/Users/suomimar/Library/Developer/CoreSimulator/tiles"
+            // This is for Android simulator
+            tileCachePath="/data/user/0/com.airbnb.android.react.maps.example/files/tiles"
```

Comment:
@MaRaSu Are we going to rely developer to figure out the location of where the file should be saved? It might make sense to configure the file path on the native side.

Reply:
@christopherdro yes, the current implementation expects developer to use react-native-fs or expo-file-system or something similar to get a correct path.

In earlier version (that @felipecsl commented above) for iOS there was automatic configuration for iOS app Documents directory if` tileCachePath = ""`, however in Android empty string for property never got through to the native code. 

I'm no expert in RN mechanisms for passing properties and difference between iOS & Android so I did not dive deeper into resolving that. A workaround could be naturally to add yet another property to indicate native side to assign the file path but that does not feel elegant.

Reply:
I think leaving it up to the developer to specify the file path makes sense but am wondering if we should throw an error if a file path is invalid or not specified. Is there any use of the feature if the tiles can't be saved?

Reply:
@christopherdro few thoughts on this:
- if `tileCachePath` nor `maxNativeZ` is not specified then old UrlTile and WMSTile implementations are used and they work as normal (without caching obviously). For this case in iOS there is just few lines of new code in the execution path before "falling back" to old implementation, in Android there is slightly more but the principle is the same - the purpose here is 2 fold: ensure full compatibility with existing clients of RNM as well as de-risk this new implementation: new code only gets activated with new properties.
- RNM does not seem to throw errors easily e.g. old implementation of URLTile nor WMSTile do not throw errors if absolutely mandatory property URLTemplate is missing - therefore I did not consider throwing errors for any errors in properties either e.g. for malformed / invalid `tileCachePath,` however some defensive code has been built in.

Reply:
@MaRaSu Thank you for the explanation. I was able to successfully test on iOS and will try and test Android over the weekend. Would you be willing to update the docs for our tile components following the same format as the [others](https://github.com/react-native-maps/react-native-maps/blob/master/docs/).

Reply:
@christopherdro good to hear about the progress with testing. Let me know if any questions arise from testing or code review.

I pushed a commit for the new Docs page for `URLTile`and `WMSTile`. Might benefit from some additional text explaining or referencing some link for map tiles basics, as reading through RNM issues log gives me a feeling that many developers just want to enable (free) OpenStreetMap while do not really understand the basics of map tiling and tileservers.

Reply:
Yea I noticed that as well. What you put together is great. Thanks again!